scene/main/node.cpp

```diff
@@ -1770,6 +1770,16 @@ void Node::replace_by(Node* p_node,bool p_keep_data) {
 		}
 	}
 
+	Node* current_scene = NULL;
+#ifdef TOOLS_ENABLED
+	if (get_tree()->is_editor_hint())
+		current_scene = get_tree()->get_edited_scene_root();
+	else
+#endif
+    current_scene = get_tree()->get_current_scene();
```

Comment:
The sad face is because that line was using lines instead of tabs. BTW, I am rebasing again. This should not be needed with your method, dunno what I was thinking :laughing: